app/policy.rb

```diff
@@ -43,15 +45,19 @@ def allocatable_subnets(reserved)
   # Allocate an IP address from within the given set of available addresses.
   # Returns nil if no available addresses
   #
-  # @param addresses [Array<IPAddr>] available host addresses
+  # @param pol [AddressPool] pool to allocate from
   # @return [IPAddr] or nil
-  def allocate_address(addresses)
-    if addresses.empty?
-      nil
-    elsif addresses.size > 100
-      addresses[rand(0..100)]
+  def allocate_address(pool)
+    available = pool.available_addresses.first(100).to_a
+
+    if available.empty?
+      warn "Address pool=#{pool.id} allocates range=#{pool.allocation_range} with no available addresses"
+
+      return nil
     else
-      addresses[0]
+      info "Address pool=#{pool.id} allocates from range=#{pool.allocation_range} with available=#{available.size}#{available.size >= 100 ? '+' : ''} addresses"
+
+      return available[rand(0...available.size)]
```

Comment:
Fixed to use [`Array#sample`][http://ruby-doc.org/core-1.9.3/Array.html#method-i-sample)